### PR TITLE
Add optional Binding HTTP response body into spec

### PR DESCRIPTION
**Pull request by RicardoNiepel**
# Description

If an input binding is invoked, the applicaiton can return a HTTP response body to let Dapr automatically send data to a state store or to a output binding.

If multiple output bindings are specified, it is possible to configure if the data is send in parallel or sequential. 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #214

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

**Comment by RicardoNiepel** [comment]
@yaron2 pls review

File: reference/api/bindings_api.md [hunk 1]
```diff
@@ -80,6 +80,73 @@ def incoming():
     return "Kafka Event Processed!"
 ```
 
+### Binding subscription
+
+On startup Dapr will invoke the following endpoint on user code to discover binding subscriptions:
```
Comment (amanbha) [inline reference/api/bindings_api.md]: This needs rewording: Bindings are discovered from component yamls. Dapr calls this endpoint on startup to ensure that app indeed is handling it, if the app doesnt have the endpoint, dapr ignores it.

File: reference/api/bindings_api.md [hunk 2]
```diff
@@ -80,6 +80,73 @@ def incoming():
     return "Kafka Event Processed!"
 ```
 
+### Binding subscription
```
Comment (amanbha) [inline reference/api/bindings_api.md]: Binding subscription doesn't seem the correct term here. It has potential to confuse customers with pubsub.

Comment (msfussell) [reply reference/api/bindings_api.md]: I addressed the comments from Aman. Take a look

**Pull request merged**